src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -39,22 +40,28 @@ public AddCommand parse(String args) throws ParseException {
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_NOK,
                         PREFIX_DESCRIPTION, PREFIX_TAG);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_NOK,
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL,
                 PREFIX_DESCRIPTION)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
 
         argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_NOK,
                 PREFIX_DESCRIPTION);
+        // Compulsory fields
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
         Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
         Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
         Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
         Description description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION).get());
-        NextOfKin nextOfKin = ParserUtil.parseNextOfKin(argMultimap.getValue(PREFIX_NOK).get());
         Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
 
+
+        Optional<NextOfKin> nextOfKin = Optional.ofNullable(null);
+        if (argMultimap.getValue(PREFIX_NOK).isPresent()) {
```

Comment:
use `map`, so you can get rid of the if statement and just do NextOfKin nextOfKin = argMultimap.getValue(PREFIX_NOK).map(ParserUtil::parseNextOfKin);

Reply:
If I use map like that I have to handle some parse exception.
Tried with ifPresent and map and having the same issue
<img width="963" alt="image" src="https://github.com/AY2324S2-CS2103T-W12-4/tp/assets/48405651/26cdf14f-3ef1-429e-82eb-dc5e9c840b20">